versioned_hdf5/tests/test_api.py

```diff
@@ -2845,7 +2865,7 @@ def test_other_compression_bad_value(tmp_path, library):
     path = tmp_path / "tmp.h5"
     with h5py.File(path, "w") as f:
         vf = VersionedHDF5File(f)
-        with vf.stage_version("r0") as sv, pytest.raises(ValueError, match="invalid"):
+        with pytest.raises(ValueError, match="invalid"), vf.stage_version("r0") as sv:
```

Comment:
This PR causes validation of compression to happen upon context exit instead of immediately when calling `create_dataset`. I don't think anybody cares about this change?

Reply:
I think that's fine, as long as the same exception is raised.